pyrule_compendium/__init__.py

```diff
@@ -102,4 +113,31 @@ def get_image(self, entry: types.entry) -> objects.entry_image:
             - type: `objects.entry_image`
         """
 
-        return objects.entry_image(self.get_entry(entry), self.api)
\ No newline at end of file
+        if self._is_master_mode_entry(entry): return objects.entry_image(self.get_entry(entry), self.master_api)
+        else: return objects.entry_image(self.get_entry(entry), self.api)
+
+    def _is_master_mode_entry(self, entry: types.entry, timeout: types.timeout=None):
+        """
+        Determines if an entry is from master mode or not.
+
+        Parameters:
+            * `entry`: The ID or name of the entry.
+                - type: str, int
+            * `timeout`: Seconds to wait for response until raising `requests.exceptions.ReadTimeout`
+                - default: `compendium.default_timeout`
+                - type: int, float, tuple (for connect and read timeouts)
+
+        Returns: Whether an entry is from master mode or not
+            - type: `objects.entry_image`
+        """
+
+        if not timeout:
+            timeout = self.default_timeout
+
+        res: dict = self.api.request(f"/entry/{entry}", timeout)
+        if res: return False
+
+        res = self.master_api.request(f"/entry/{entry}", timeout)
```

Comment:
Too many requests?

Reply:
Again, I don’t see a way to minimize the number of requests we have here other than using local data.